### PR TITLE
Adjust @GuardedBy to pass internal GuardedBy Checking.

### PR DESCRIPTION
**Pull request by madongfly**
This is required by our internal sync.

@ejona86 please take a look, thanks!

### CODE REVIEW

**Comment by madongfly** [comment]
And maybe we should start using [error prone](https://github.com/google/error-prone)?

File: okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java [hunk 1]
```diff
@@ -419,7 +419,11 @@ ClientFrameHandler getHandler() {
     return clientFrameHandler;
   }
 
+  /**
```
Comment (ejona86) [inline okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java]: Can you just return a copy?

**Comment by ejona86** [comment]
@madongfly, I'd be happy if we had error-prone integrated. I'd mainly want to make sure it has good signal vs noise; I seem to remember it has a lot of possible things it can check, some great, some noisy.

File: okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java (hunk 1 above)
Comment (madongfly) [inline okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java]: Removed it after updated the test.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

**Comment by ejona86** [comment]
@madongfly LGTM

**Pull request merged**